src/Streams.sol

```diff
@@ -683,29 +736,29 @@ abstract contract Streams {
     /// @param newReceivers The list of the streams receivers of the account to be set.
     /// Must be sorted by the account IDs and then by the stream configurations,
     /// without identical elements and without 0 amtPerSecs.
-    /// @param maxEndHint1 An optional parameter allowing gas optimization, pass `0` to ignore it.
-    /// The first hint for finding the maximum end time when all streams stop due to funds
+    /// @param maxEndHints An optional parameter allowing gas optimization.
+    /// Pass a list of 8 zero value hints to ignore it, it's represented as an integer `0`.
+    /// The list of hints for finding the maximum end time when all streams stop due to funds
     /// running out after the balance is updated and the new receivers list is applied.
     /// Hints have no effect on the results of calling this function, except potentially saving gas.
     /// Hints are Unix timestamps used as the starting points for binary search for the time
     /// when funds run out in the range of timestamps from the current block's to `2^32`.
-    /// Hints lower than the current timestamp are ignored.
-    /// You can provide zero, one or two hints. The order of hints doesn't matter.
+    /// Hints lower than the current timestamp including the zero value hints are ignored.
+    /// If you provide fewer than 8 non-zero value hints make them the rightmost values to save gas.
+    /// It's the best approach to make the most risky and precise hints the rightmost ones.
```

Comment:
I think it would good to mention specifically that there are no assumptions.
- no specific ordering of the hints is strictly required
  - only the already mentioned gas improvement with risky hints on the right
- all hints could be used to improve either higher bound or the lower bound 
- it is not required that hints come in pairs of two with a start and end
- provided list only improves gas usage and has no impact on the outcome of the transaction

I think a potentially confusing part is: why is it a list with multiple hints. 

Maybe a few strategies to illustrate the concept. Alternatively a link to some documentation

For example: Unkown block.timstamp of the new transactions together with streams which should stop when running out of funds:

- risky hint pair a `1 min` start/end range
- 5 minute hint pair start/end range
- 2 hours pair for the worst case

Reply:
Clarified.